src/makeStyles.tsx

```diff
@@ -19,14 +19,17 @@ let counter = 0;
 
 export function createMakeStyles<Theme>(params: {
     useTheme: () => Theme;
-    cache?: EmotionCache;
+    cache?: EmotionCache | (() => EmotionCache);
 }) {
-    const { useTheme, cache } = params;
+    const { useTheme, cache: cacheOrGetCache } = params;
 
     function useCache() {
         const contextualCache = useContextualCache();
 
-        const cacheToBeUsed = cache ?? contextualCache;
+        const cacheToBeUsed =
+            typeof cacheOrGetCache === "function"
+                ? cacheOrGetCache()
+                : cacheOrGetCache ?? contextualCache;
```

Comment:
@ryancogswell Can you confirm this what we want? The callback is called every render.